### PR TITLE
CA-366706: Clean up signal handling

### PR DESCRIPTION
**Pull request by rosslagerwall**

### CODE REVIEW

File: varstored.c [hunk 1]
```diff
@@ -330,14 +330,10 @@ varstored_teardown(void)
...
 varstored_sigterm(int num)
 {
-    INFO("%s\n", strsignal(num));
-
-    varstored_teardown();
-
-    if (num == SIGTERM && run_main_loop)
+    if (run_main_loop)
         run_main_loop = 0;
     else
```
[the remark this answers is not in the document]
Comment (lindig) [reply varstored.c:335]: Is the name `varstored_sigterm` still correct or is this now handling other signals as well?

[the remark this answers is not in the document]
Comment (rosslagerwall) [reply varstored.c:335]: > Is that acceptable or should there be another boolean here that reflects the running state of the main loop?

I think it is acceptable. It's an edge case without any serious consequences so probably not worth handling it.

[the remark this answers is not in the document]
Comment (rosslagerwall) [reply varstored.c:335]: > Is the name varstored_sigterm still correct or is this now handling other signals as well?

It's always been incorrectly named... I've added an extra commit to fix that.

**Review by lindig — APPROVED** [review]

**Review by edwintorok — APPROVED** [review]

**Comment by lindig** [comment]
@rosslagerwall who can review this for a merge?

**Pull request merged**